site/src/load.rs

```diff
@@ -199,6 +222,16 @@ impl SiteCtxt {
             all_commits,
         )
     }
+
+    /// Download master-branch Rust commits if the cached value is older than one minute
+    pub async fn update_master_commits(&self) -> anyhow::Result<()> {
+        let commits = self.master_commits.load();
+        if commits.updated.elapsed() > std::time::Duration::from_secs(60) {
```

Comment:
Ideally we'd probably return results immediately always (if available) and then start a fetch in the background if it's time to update - stale results don't really hurt us much. It might be annoying to make sure there's only one background fetch running though, not sure.

Otherwise I suspect this wouldn't help much since many requests happen more than 60 seconds apart, I'd guess

Reply:
I have implemented this. I don't think we need to bother making sure there's only one background fetch at a time. Worst case if N people request within ~half a second, we fetch N times, as we would without caching at all.